src/components/AuditLogsTable/AuditLogsTable.test.tsx

```diff
@@ -1,40 +1,128 @@
+import { screen, within } from "@testing-library/react";
+import { add } from "date-fns";
+
+import { TestId } from "components/LoadingSpinner/LoadingSpinner";
+import { actions as jujuActions } from "store/juju";
+import * as storeModule from "store/store";
+import type { RootState } from "store/store";
+import { rootStateFactory } from "testing/factories";
+import { configFactory, generalStateFactory } from "testing/factories/general";
+import { auditEventFactory } from "testing/factories/juju/jimm";
 import { renderComponent } from "testing/utils";
 
 import AuditLogsTable from "./AuditLogsTable";
 
 describe("AuditLogsTable", () => {
-  it("should have model as second header when showing the model data", () => {
-    renderComponent(<AuditLogsTable showModel={true} />);
-    expect(document.querySelector("table tr")).toBeVisible();
-    expect(document.querySelectorAll("table th")[1]).toHaveTextContent("model");
+  let state: RootState;
+
+  beforeEach(() => {
+    const dispatch = jest
+      .fn()
+      .mockImplementation(() =>
+        Promise.resolve({ events: [auditEventFactory.build()] })
+      );
+    jest.spyOn(storeModule, "usePromiseDispatch").mockReturnValue(dispatch);
+    state = rootStateFactory.build({
+      general: generalStateFactory.build({
+        config: configFactory.build({
+          controllerAPIEndpoint: "wss://example.com/api",
+        }),
+        controllerConnections: {
+          "wss://example.com/api": {
+            user: {
+              "display-name": "eggman",
+              identity: "user-eggman@external",
+              "controller-access": "",
+              "model-access": "",
+            },
+          },
+        },
+      }),
+    });
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("should have model as second header when showing the model data", async () => {
+    renderComponent(<AuditLogsTable showModel={true} />, { state });
+    expect((await screen.findAllByRole("columnheader"))[1]).toHaveTextContent(
```

Comment:
Once I rebased I had to update a number of tests to fetch the audit logs and wait for the promise to resolve.